protocol/0078-NWLI-network_wide_limits.md

```diff
@@ -0,0 +1,128 @@
+
+* if the order is aggressive (results in a trade) the trade proceeds as normal,
+* if the order or its part is passive (it doesn't match in full on entry to the matching engine) and it's part would get added to the order book (it's a persistent order) that part of the order gets rejected.
+
+Pegged orders and [liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total number of orders on the book in any given market then no further actions are needed. Specifically: all open orders in that market remain unaffected, but no new orders are allowed until the **total open order count** drops below the new limit.
+
+## Pegged orders on a market
+
+Introduce a new network parameter `limits.markets.maxPeggedOrders` controlling the maximum number of pegged orders that can be active in any given market.\
+Default value: `10,000`.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.\
+If it gets decreased below the current total number of orders on the book in any given market then no further actions are needed. Specifically: all pegged orders already present in that market remain unaffected, but no new pegged orders are allowed until the **total pegged order count** drops below the new limit.
+
+[Liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
+
+## LP order shapes
+
+Each [LP order shape](./0038-OLIQ-liquidity_provision_order_type.md#how-they-are-submitted) has a hardcoded limit of at most 20 entries (offsets).
+
+## LPs in a market
+
+Introduce a new network parameter `limits.markets.maxLPs` controlling the maximum number of liquidity providers that can be active in any market.\
+Default value: `1000`.
+
+Each liquidity provider that successfully submits a liquidity provision transaction for a given market gets counted towards the **active LP count** for that market. When an LP in a given market reduces their commitment amount to 0 or gets closed out the LP count for that market gets decremented by 1.
+
+Once the active LP count in a given market reaches the limit only the LP transactions with commitment amount larger then the lowest commitment already active in that market are accepted. If such a situation occurs the existing LP commitment with the lowest value which has been in the market the shortest should get forcibly cancelled.\
```

Comment:
I am not sure we want that. Imagine I was the first LP on the market; small commitment but *huge* ELS. This would be a way for someone to force me out.

Reply:
Good point, committed your suggestion.